gtsfm/densify/interface_patchmatchnet.py

```diff
@@ -0,0 +1,182 @@
+"""Interface class from gtsfmData to patchmatchnetData
+
+Authors: Ren Liu
+"""
+from typing import Any, Dict, List, Tuple
+
+import cv2
+import numpy as np
+from torch.utils.data import Dataset
+
+from gtsfm.common.image import Image
+from gtsfm.common.gtsfm_data import GtsfmData
+from gtsfm.densify.mvs_math import piecewise_gaussian
+
+NUM_PATCHMATCHNET_STAGES = 4
+
+
+class PatchmatchNetData(Dataset):
+    """PatchmatchNetData class for Patchmatch Net. It contains the interface from GtsfmData.
+    Wang's work in https://github.com/FangjinhuaWang/PatchmatchNet/blob/main/datasets/dtu_yao_eval.py is referred to.
+    """
+
+    def __init__(self, images: Dict[int, Image], sfm_result: GtsfmData, num_views: int = 5) -> None:
+        """Initialize method for PatchmatchnetData
+
+        Args:
+            images: input images (H, W, C) to GTSFM
+            sfm_result: sfm results calculated by GTSFM
+            num_views: number of views, containing 1 reference view and (num_views-1) source views
+        """
+        assert images is not None and len(images) > 1
+
+        # cache sfm result
+        self._sfm_result = sfm_result
+
+        # Patchmatch Net meta
+        self._num_views = num_views
+        self._num_stages = NUM_PATCHMATCHNET_STAGES
+
+        # Test data preparation
+        self._keys = sorted(self._sfm_result.get_valid_camera_indices())
+        self._num_images = len(self._keys)
+        self._keys_map = {}
+        for i in range(self._num_images):
+            self._keys_map[self._keys[i]] = i
```

Comment:
is there a clearer name than "keys_map"? what does this map represent?

Reply:
if self._keys = [2,4,5,7,8], then the self._keys_map defines the position of a specific key in the key list, like self._keys_map[2] = 0

Reply:
agreed w/ Ayush. inline definitions of what those mean would help too